java/ql/test/experimental/query-tests/security/CWE-074/XsltInjection.java

```diff
@@ -0,0 +1,122 @@
+import java.io.InputStreamReader;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.net.Socket;
+import java.net.URI;
+
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.transform.Source;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.sax.SAXSource;
+import javax.xml.transform.stax.StAXSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+
+import org.springframework.web.bind.annotation.RequestParam;
+import org.xml.sax.InputSource;
+
+import net.sf.saxon.s9api.Processor;
+import net.sf.saxon.s9api.XdmValue;
+import net.sf.saxon.s9api.XsltCompiler;
+
+public class XsltInjection {
+  public void testStreamSourceInputStream(Socket socket) throws Exception {
+    StreamSource source = new StreamSource(socket.getInputStream());
+    TransformerFactory.newInstance().newTransformer(source).transform(null, null);
+  }
+
+  public void testStreamSourceReader(Socket socket) throws Exception {
+    StreamSource source = new StreamSource(new InputStreamReader(socket.getInputStream()));
+    TransformerFactory.newInstance().newTemplates(source).newTransformer().transform(null, null);
+  }
+
+  public void testStreamSourceInjectedParam(@RequestParam String param) throws Exception {
```

Comment:
Modelling of taint sources based on `@RequestParam` has been improved a bit, so these methods also need a `@RequestMapping` annotation in order for this to be considered a taint source (the test is currently failing due to this).

Reply:
I've added `@RequestMapping` annotation to the methods with `@RequestParam`.
However, the test was passing for me even without this annotation.